src/core/ActivityInfoTree.cpp

```diff
@@ -283,5 +285,71 @@ void ActivityInfoTree::init()
 
     qmlRegisterSingletonType<QObject>("GCompris", 1, 0, "ActivityInfoTree", menuTreeProvider);
     qmlRegisterType<ActivityInfo>("GCompris", 1, 0, "ActivityInfo");
+
+
+}
+void ActivityInfoTree::filterBySearch(const QString& text)
+{
+
+    if(text.trimmed().length() < m_searchedtext.length() or text.trimmed() == ""){
```

Comment:
the current code will not display any activity if you delete 'e' and type 'a' but  if you delete 
'a' and type 'e' again in this case no activity will be displayed although all the activities with the name 'chess' should have been displayed .So yes there is issue with the current code but I have come up with a solution to handle such cases so if it's okay we can keep doing the optimization with handling the case that you  have mentioned.

Reason why I would like to keep the optimized code:

1. If the user is using virtual keyboard there is no way for him to go back and delete a specific character   that's because we disable the focus if user has opted to use  virtual  keyboard and if there is no focus the cursor won't be visible .

2. ex: if I type "planefame"  instead of "planegame"  I would delete 'fame' not just 'f' and type 'game'    ,this I what I usually do  unless I  typed a very long string and made a mistake of mistyping a single character but we can't expect every user to do that .Many would prefer deleting that single mistyped character and my new approach handles this case :)